stdlib/src/collections/list.mojo

```diff
@@ -91,11 +99,19 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
 
     fn __init__(inout self):
         """Constructs an empty list."""
-        self.data = UnsafePointer[T]()
+        self._small_buffer = Self._small_buffer_type(uninitialized=True)
         self.size = 0
-        self.capacity = 0
+        # needed to avoid "potential indirect access to uninitialized value 'self.data'"
+        self.data = UnsafePointer[T]()
```

Comment:
Could you move this into the else clause? Trying to avoid writing to self.data twice.